registrations/views.py

```diff
@@ -43,6 +52,24 @@ def get_initial(self):
         return super().get_initial()
 
 
+class RegistrationConfirmOptIn(TemplateView):
+    template_name = "registrations/confirm_optin.html"
+
+    def dispatch(self, request, *args, **kwargs):
+        if (
+            "registration_details" not in request.session
+            or "msisdn" not in request.session.get("registration_details", {})
+        ):
+            return redirect(reverse_lazy("registrations:registration-details"))
+        return super().dispatch(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        if "yes" in request.POST:
+            request.session["channel"] = "WhatsApp"
```

Comment:
I'm not sure why we're setting the channel here? The `confirm-clinic` view should set the channel for us.

Reply:
whoops